new-dti-website/components/apply/ApplicationTimeline.tsx

```diff
@@ -0,0 +1,270 @@
+import { useState } from 'react';
+import Image from 'next/image';
+import config from '../../config.json';
+import timelineIcons from './data/timelineIcons.json';
+import { ibm_plex_mono } from '../../src/app/layout';
+
+type TabProps = {
+  isSelected: boolean;
+  text: string;
+  onClick?: () => void;
+};
+
+const Tab: React.FC<TabProps> = ({ isSelected, text, onClick }) => (
+  <div
+    className={`flex items-center lg:px-5 lg:py-4 md:px-4 md:py-3 xs:px-2 md:rounded-t-xl xs:rounded-t-lg ${
+      isSelected ? 'bg-[#FEFEFE] text-[#A52424]' : 'bg-[#7E2222CC] text-[#FEFEFE]'
+    } hover:cursor-pointer md:h-min xs:h-full`}
+    onClick={onClick}
+  >
+    <p className="font-bold lg:text-lg md:text-[13px] xs:text-[10px]">{text}</p>
+  </div>
+);
+
+type DateTime = {
+  date: string;
+  isTentative: boolean;
+  time?: string;
+};
+
+type RecruitmentEvent = {
+  title: string;
+  description: string;
+  location?: string;
+  type: string;
+  link?: string;
+  freshmen: DateTime;
+  upperclassmen: DateTime;
+  spring: DateTime;
+};
+
+/**
+ * Find the end time of an DateTime as a number, regardless if the event is
+ * tentative or not. If no time is specified, assume the end time is the last
```

Comment:
Thanks for the docstring! I think specifying the format of the input date and time would be helpful for readability. Curious also if we could make this simpler with built-in `Date` functions.